### PR TITLE
Tradingapi

### LINKED ISSUE #134: Updating Trading Ideas to Use data from APIs
Update Trading ideas to show data fetched from APIs.

### PR DESCRIPTION
**Pull request by anagarwa**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #134 

Test URLs:
- Before: https://main--icicidirect--aemsites.hlx.live/research/equity
- After: https://tradingapi--icicidirect--aemsites.hlx.live/research/equity

### CODE REVIEW

File: blocks/multi-panel-carousel/multi-panel-carousel.js [hunk 1]
```diff
@@ -336,13 +340,55 @@ function getRecommendationsCard(companies, type) {
...
+    inputJson: JSON.stringify({
+      rating: '1', timeFrame: '', pageNo: '1', pageSize: '5',
+    }),
+  };
+
+  postFormData(getResearchAPIUrl(), jsonFormData, (error, tradingData = []) => {
+    if (error === null) {
+      const recommendationArray = tradingData.Data.Table;
+      const companiesArray = [];
+      recommendationArray.forEach((company) => {
```
Comment (JiangLong2019) [inline blocks/multi-panel-carousel/multi-panel-carousel.js:368]: Maybe we can involve `parseResponse()` method in block utils

File: blocks/multi-panel-carousel/multi-panel-carousel.js [hunk 2]
```diff
@@ -427,7 +473,7 @@ function addDiscoverLink(carouselBody, discoverLink) {
 }
 
 function getHighlightDiv(block) {
-  const predicationDiv = block.querySelectorAll(':scope > div')[2].children[1];
+  const predicationDiv = block.querySelectorAll(':scope > div')[3].children[1];
```
Comment (sdmcraft) [inline blocks/multi-panel-carousel/multi-panel-carousel.js:476]: This makes hard assumptions on a block to be authored in a particular way like "the 4th row in the block must be this or that" . Can we employ a method like `getBlockElements(key)` in which you give the `key` and get the corresponding markup div in response.

Comment (anagarwa) [reply blocks/multi-panel-carousel/multi-panel-carousel.js:476]: @sdmcraft yes that has been discussed in the pr https://github.com/aemsites/icicidirect/pull/137.

Comment (sdmcraft) [reply blocks/multi-panel-carousel/multi-panel-carousel.js:476]: I'd suggest we have a simplified `getBlockElements` in #137 and merge it and then use that here.

Comment (anagarwa) [reply blocks/multi-panel-carousel/multi-panel-carousel.js:476]: @sdmcraft updated pr #137 to have a simple blockMarkUp reader. Once that is merge, I will remove these changes from here

Comment (anagarwa) [reply blocks/multi-panel-carousel/multi-panel-carousel.js]: PR #137 has been merged and update this pr basis on that

**Review by ravverma — APPROVED** [review]
LGTM,
Just one thing that Discover more arrow icon is still missing.

**Comment by anagarwa** [comment]
> LGTM, Just one thing that Discover more arrow icon is still missing.

@ravverma opened https://github.com/aemsites/icicidirect/issues/156 as a follow up

**Pull request merged**